### PR TITLE
Support for IPv4 Subnets

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
Support for IPv4 Subnets

### Check List
- [ ] New functionality includes sanity testing.
  - [ ] All sanity tests pass. (`$ ansible-test sanity`).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).

### CODE REVIEW

**Review by SHANDCRUZ — APPROVED** [review]

File: plugins/modules/oneview_id_pools_ipv4_subnet.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    MSG_COLLECT = 'Collected the ids allocated'
+    RESOURCE_FACT_NAME = 'id_pools_ipv4_subnet'
+
+    def __init__(self):
+
+        additional_arg_spec = dict(data=dict(required=True, type='dict'),
+                                   state=dict(required=True,
+                                   choices=['present', 'absent', 'allocate', 'collect']))
+
+        super(IdPoolsIpv4SubnetModule, self).__init__(additional_arg_spec=additional_arg_spec,
```
Comment (VenkateshRavula) [inline plugins/modules/oneview_id_pools_ipv4_subnet.py:102]: Try not to five arguments to super() class in python3

File: roles/oneview_id_pools_ipv4_subnet/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+| validate_etag  |   |  True  | <ul> <li>true</li>  <li>false</li> </ul> |  When the ETag Validation is enabled, the request will be conditionally processed only if the current ETag for the resource matches the ETag provided in the data.  |
+
+## Example Playbook
+
+```yaml
+- hosts: all
+  collections:
+    - hpe.oneview
+  roles:
+    - hpe.oneview.oneview_id_pools_ipv4_subnets
```
Comment (VenkateshRavula) [inline roles/oneview_id_pools_ipv4_subnet/README.md:26]: role name should be singular

File: roles/oneview_id_pools_ipv4_subnet/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## License
+
+Apache
+
+#### Return Values
+
+| Name          | Description  | Returned | Type       |
+| ------------- |-------------| ---------|----------- |
+| id_pools_ipv4_subnet | Has the facts about the managed OneView ID Pools IPv4 Subnet |  On state 'present'. Can be null. |  dict |
```
Comment (VenkateshRavula) [inline roles/oneview_id_pools_ipv4_subnet/README.md:37]: As per the library files, return value is ```id_pools_ipv4_subnets```

File: roles/oneview_id_pools_ipv4_subnet/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+# defaults file for oneview_id_pools_ipv4_subnets
+config: "~/.ansible/collections/ansible_collections/hpe/oneview/roles/oneview_id_pools_ipv4_subnets/files/oneview_config.json"
```
Comment (VenkateshRavula) [inline roles/oneview_id_pools_ipv4_subnet/defaults/main.yml:3]: check the path properly. You have mentioned the role name as ```oneview_id_pools_ipv4_subnet``` singular but the path to config file is showing ```oneview_id_pools_ipv4_subnets``` plural

File: roles/oneview_id_pools_ipv4_subnet/meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+---
+galaxy_info:
+  role_name: oneview_id_pools_ipv4_subnets
```
Comment (VenkateshRavula) [inline roles/oneview_id_pools_ipv4_subnet/meta/main.yml:3]: change role name to singular

File: tests/unit/test_oneview_id_pools_ipv4_subnet.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+# limitations under the License.
+###
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import mock
+import pytest
+
+from hpe_test_utils import OneViewBaseTest
```
Comment (VenkateshRavula) [inline tests/unit/test_oneview_id_pools_ipv4_subnet.py:24]: check the path of ```hpe_test_utils``` in remaining UT files and change accordingly

**Review by VenkateshRavula — APPROVED** [review]

**Pull request merged**